src/main/java/core/basesyntax/impl/PurchaseHandlerImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.impl;
+
+import dao.FruitDao;
+import dao.FruitDaoImpl;
+import model.FruitTransaction;
+import strategy.OperationHandler;
+
+public class PurchaseHandlerImpl implements OperationHandler {
+    private FruitDao fruitDao = new FruitDaoImpl();
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        int quantityBefore = fruitDao.get(transaction.getFruit());
+        if (quantityBefore < transaction.getQuantity()) {
```

Comment:
The error message in the RuntimeException should be more informative. Include details about which fruit and how much quantity was attempted to be purchased versus the available amount in the storage.